platforms/core-runtime/stdlib-java-extensions/src/main/java/org/gradle/util/internal/CollectionUtils.java

```diff
@@ -621,52 +609,6 @@ public static <T> Pair<Collection<T>, Collection<T>> partition(Iterable<T> items
         return Pair.of(left, right);
     }
 
-    public static class InjectionStep<T, I> {
-        private final T target;
-        private final I item;
-
-        public InjectionStep(T target, I item) {
-            this.target = target;
-            this.item = item;
-        }
-
-        public T getTarget() {
-            return target;
-        }
-
-        public I getItem() {
-            return item;
-        }
-    }
-
-    public static <T, I> T inject(T target, Iterable<? extends I> items, Action<InjectionStep<T, I>> action) {
-        if (target == null) {
-            throw new NullPointerException("The 'target' cannot be null");
-        }
-        if (items == null) {
-            throw new NullPointerException("The 'items' cannot be null");
-        }
-        if (action == null) {
-            throw new NullPointerException("The 'action' cannot be null");
-        }
-
-        for (I item : items) {
-            action.execute(new InjectionStep<T, I>(target, item));
-        }
-        return target;
-    }
-
-    public static <K, V> Map<K, Collection<V>> groupBy(Iterable<? extends V> iterable, InternalTransformer<? extends K, V> grouper) {
```

Comment:
💭  There was only one call to this in a Java 8-capable module, where JDK Streams can be used